src/Yaapii.Atoms/IO/TempDirectory.cs

```diff
@@ -0,0 +1,66 @@
+﻿// MIT License
+//
+// Copyright(c) 2017 ICARUS Consulting GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.IO;
+
+namespace Yaapii.Atoms.IO
+{
+    /// <summary>
+    /// A directory that cleans up on dispose.
+    /// </summary>
+    public sealed class TempDirectory : IScalar<DirectoryInfo>, IDisposable
+    {
+        private readonly string path;
+
+        public TempDirectory(string path)
```

Comment:
Didnt see it in last reciew. Request was to get a temporary path with Path.GetTempPath() + maybe a random number or a guid.

So user is not forced to inject a path. Its more usable when you do not need to care about what path it uses.

Injecting a path can stay but must be optional.